app/pages/admin/project-status.jsx

```diff
@@ -257,6 +324,52 @@ class ProjectStatus extends Component {
                 </label>
               </div>
               <hr />
+              <div>
+                <h4>Configure Training Data</h4>
+                <label>
+                  Training Set IDs:{' '}
+                  <input
+                    id="training-set-ids"
+                    type="text"
+                    onBlur={(event) => this.onBlurTrainingSetIds(workflow, event)}
+                    onChange={(event) => {
+                      const updatedTrainingSetIds = Object.apply({}, this.state.valTrainingSetIds)
```

Comment:
Should this be `Object.assign` in each of the change handlers? [`apply`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) is used to call a function with an explicit `this` context.

```suggestion
                      const updatedTrainingSetIds = Object.assign({}, this.state.valTrainingSetIds)
```